### DISCUSSION TITLE
Is there a way to generate a class with a List of objects?

### DISCUSSION THREAD

**Discussion by fleeser** [serverpod/serverpod#685] [Q&A] [answered]
I found out that I am allowed to define classes with fields that contains `Lists` of type `String`, `int` and so on.

```
class: UserResponse
fields:
  user: List<int>?
```

Also I can define classes that hold fields of type `Object`.

```
class: UserResponse
fields:
  user: PublicUser?
```

But when trying to define classes that hold fields of type `List<Object>`,

```
class: UserResponse
fields:
  user: List<PublicUser>?
```

the generated file shows the following error:

`import 'protocol.dart' as _i2;`

> Target of URI doesn't exist: 'protocol.dart'.
> Try creating the file referenced by the URI, or try using a URI for a file that does exist.

**Comment by yahu1031** [comment, marked as answer]
yeah, check the `protocol.dart` file path. That wasn't mapping correctly sometimes. Did you use any subdirectories in `protocol`? If yes, here is the related ticket - #675, #624(PR), #546(PR)

**Reply by fleeser** [reply]
#622 describes exactly that behavior I am facing, but it was fixed by @fischerscode and merged into main 2 weeks ago. But it did not make it into release 1.0.1, right?

**Reply by yahu1031** [reply]
It did. Here it is serverpod_cli v1.0.1 - [ChangeLog](https://pub.dev/packages/serverpod_cli/changelog#101)

**Reply by fleeser** [reply]
Tried with the updated version, but still does not fix this issue. When adding user_response.yaml to the root dir of the protocol folder, it gets generated correctly.

Structure looks like this:

- protocol
    - user
        - user_response.yaml

**Reply by fischerscode** [reply]
Please verify that you are using version `1.0.1` by running `dart pub global list`.
The output should contain `serverpod_cli 1.0.1`.

**Reply by fleeser** [reply]
Problem was that I updated the version inside the `pubspec.yaml` file, but not the serverpod_cli.
